### DISCUSSION TITLE
Not able to update LIVE question set

### DISCUSSION THREAD

**Discussion by siddhishinde0723** [Sunbird-Ed/Community#676] [General Discussion]
5.0 is the current Sunbird release version.

We are attempting to edit the currently active question set, but we are encountering internal server problems. We are trying to edit the question set which are live but we are getting internal server issue. 

1. CURL:
`curl --location --request PATCH 'https://sunbirdsaas.com/action/questionset/v1/add' \
--header 'Accept: application/json' \
--header 'Accept-Language: en-GB,en;q=0.9' \
--header 'Authorization;' \
--header 'Connection: keep-alive' \
--header 'Content-Type: application/json' \
--header 'Origin: https://sunbirdsaas.com' \
--header 'Referer: https://sunbirdsaas.com/workspace/edit/QuestionSet/do_113947291794022400152/allcontent/Live' \
--header 'Sec-Fetch-Dest: empty' \
--header 'Sec-Fetch-Mode: cors' \
--header 'Sec-Fetch-Site: same-origin' \
--header 'User-Agent: Mozilla/5.0 (X11; Linux x86_64) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/119.0.0.0 Safari/537.36' \
--header 'X-App-Id: production.production.portal' \
--header 'X-Device-ID: 811273fe91589aaedb5fddcd63a78e9b' \
--header 'X-Source: web' \
--header 'X-msgid: 21f62eb9-a187-0e80-d148-007dfae85eb4' \
--header 'sec-ch-ua: "Google Chrome";v="119", "Chromium";v="119", "Not?A_Brand";v="24"' \
--header 'sec-ch-ua-mobile: ?0' \
--header 'sec-ch-ua-platform: "Linux"' \
--header 'ts: 2020-12-31T16:23:27+05:30' \
--data '{
    "request": {
        "questionset": {
            "rootId": "do_113947291794022400152",
            "collectionId": "do_113947296363298816161",
            "children": [
                "do_113940256703545344132"
            ]
        }
    }
}'`

2. Issue: 
![image](https://github.com/Sunbird-Ed/Community/assets/72926048/29614ce0-ba39-46c4-bd7e-c9c8699ee8af)

3. We looked over the API logs. The logs for the same are as follows: 
`assessment-58cfcddf99-gd7lr assessment java.lang.ClassCastException: class java.util.ArrayList cannot be cast to class [Ljava.lang.String; (java.util.ArrayList and [Ljava.lang.String; are in module java.base of loader 'bootstrap')
assessment-58cfcddf99-gd7lr assessment 	at org.sunbird.managers.HierarchyManager$$anonfun$addLeafNodesToHierarchy$2.apply(HierarchyManager.scala:54)
assessment-58cfcddf99-gd7lr assessment 	at org.sunbird.managers.HierarchyManager$$anonfun$addLeafNodesToHierarchy$2.apply(HierarchyManager.scala:50)
assessment-58cfcddf99-gd7lr assessment 	at scala.util.Success$$anonfun$map$1.apply(Try.scala:237)
assessment-58cfcddf99-gd7lr assessment 	at scala.util.Try$.apply(Try.scala:192)
assessment-58cfcddf99-gd7lr assessment 	at scala.util.Success.map(Try.scala:237)
assessment-58cfcddf99-gd7lr assessment 	at scala.concurrent.Future$$anonfun$map$1.apply(Future.scala:237)
assessment-58cfcddf99-gd7lr assessment 	at scala.concurrent.Future$$anonfun$map$1.apply(Future.scala:237)
assessment-58cfcddf99-gd7lr assessment 	at scala.concurrent.impl.CallbackRunnable.run(Promise.scala:36)
assessment-58cfcddf99-gd7lr assessment 	at akka.dispatch.BatchingExecutor$AbstractBatch.processBatch(BatchingExecutor.scala:55)
assessment-58cfcddf99-gd7lr assessment 	at akka.dispatch.BatchingExecutor$BlockableBatch$$anonfun$run$1.apply$mcV$sp(BatchingExecutor.scala:92)
assessment-58cfcddf99-gd7lr assessment 	at akka.dispatch.BatchingExecutor$BlockableBatch$$anonfun$run$1.apply(BatchingExecutor.scala:92)
assessment-58cfcddf99-gd7lr assessment 	at akka.dispatch.BatchingExecutor$BlockableBatch$$anonfun$run$1.apply(BatchingExecutor.scala:92)
assessment-58cfcddf99-gd7lr assessment 	at scala.concurrent.BlockContext$.withBlockContext(BlockContext.scala:72)
assessment-58cfcddf99-gd7lr assessment 	at akka.dispatch.BatchingExecutor$BlockableBatch.run(BatchingExecutor.scala:91)
assessment-58cfcddf99-gd7lr assessment 	at akka.dispatch.TaskInvocation.run(AbstractDispatcher.scala:40)
assessment-58cfcddf99-gd7lr assessment 	at akka.dispatch.ForkJoinExecutorConfigurator$AkkaForkJoinTask.exec(ForkJoinExecutorConfigurator.scala:49)
assessment-58cfcddf99-gd7lr assessment 	at akka.dispatch.forkjoin.ForkJoinTask.doExec(ForkJoinTask.java:260)
assessment-58cfcddf99-gd7lr assessment 	at akka.dispatch.forkjoin.ForkJoinPool$WorkQueue.runTask(ForkJoinPool.java:1339)
assessment-58cfcddf99-gd7lr assessment 	at akka.dispatch.forkjoin.ForkJoinPool.runWorker(ForkJoinPool.java:1979)
assessment-58cfcddf99-gd7lr assessment 	at akka.dispatch.forkjoin.ForkJoinWorkerThread.run(ForkJoinWorkerThread.java:107)
assessment-58cfcddf99-gd7lr assessment Exception occurred - class :java.lang.ClassCastException with message :class java.util.ArrayList cannot be cast to class [Ljava.lang.String; (java.util.ArrayList and [Ljava.lang.String; are in module java.base of loader 'bootstrap')
assessment-58cfcddf99-gd7lr assessment 2024-01-10 06:28:14,934 {"eid":"LOG","ets":1704868094934,"ver":"3.0","mid":"LP.1704868094934.f3c0cbe5-d78f-4572-8821-9938fb501645","actor":{"id":"org.sunbird.learning.platform","type":"System"},"context":{"channel":"in.ekstep","pdata":{"id":"production.production.portal","pid":"","ver":"1.0"},"env":"assessment","sid":"a29Z-1W7dTD5gkVAduzNQ8raIY0xINTW","did":"811273fe91589aaedb5fddcd63a78e9b"},"edata":{"level":"INFO","type":"api_access","message":"","params":[{"duration":95},{"protocol":"http"},{"size":303},{"method":"PATCH"},{"uip":"10.1.9.117"},{"url":"/questionset/v4/add"},{"status":500}]},"syncts":1704868094934}`

4. We also looked at the staging, but the update is not supported there either. This is the problem I'm having there:
`{
    "id": "api.questionset.hierarchy.update",
    "ver": "5.0",
    "ts": "2024-01-11T04:28:24ZZ",
    "params": {
        "resmsgid": "c62af6a4-9c85-40bb-bc01-801f25276f75",
        "msgid": null,
        "err": "ERR_HIERARCHY_UPDATE_DENIED",
        "status": "failed",
        "errmsg": "QuestionSet can't support update hierarchy operation because it is not having data in QuML 1.1 format."
    },
    "responseCode": "CLIENT_ERROR",
    "result": {
        "messages": null
    }
}`

@vinukumar-vs @rajeevsathish @rajeshkumaravel @Krishnaj20

**Comment by vinukumar-vs** [comment]
@siddhishinde0723 as it is related to an InQuiry BB issue(assessment, Question-set etc), please move this discussion thread to[ InQuiry discussions](https://github.com/orgs/Sunbird-inQuiry/discussions/).

Meanwhile, @sajeshkayyath can you look into this issue?
